<launch>
  <include file="$(find pr2lite_moveit_config)/launch/id_server.launch" />
  <include file="$(find pr2lite_moveit_config)/launch/pc_485net_with_delay.launch" />

   <!-- Start the Dynamixel Joint States Publisher -->
  <node pkg="pr2lite_moveit_config" name="dynamixel_joint_states_publisher" type="dynamixel_joint_state_publisher.py" output="screen">
      <param name="rate" value="5" />
      <rosparam file="$(find pr2lite_moveit_config)/config/pr2lite_arm.yaml" command="load" />
  </node>

  <include file="$(find pr2lite_moveit_config)/launch/pr2lite_dynamixel.launch" />
<!--
-->

  <!-- pr2lite is an awesome mobile manipulator. 
       This launch file brings him to life. -->
<!--
  <node pkg="pr2lite_moveit_config" name="pc_485net_joints" type="pc_485net_joints.py" output="screen" />
  <node pkg="pr2lite_moveit_config" name="search_id" type="id_server.py" output="screen" >
    <param name="~uc_config_file" value="$(find pr2lite_moveit_config)/config/uc.txt" /> 
  </node>
-->
<!--
  <node pkg="pr2lite_moveit_config" name="pc_485net_joints" type="pc_485net_joints.py" output="screen" />
  <node pkg="prlite_base" name="base_controller" type="base_controller" output="screen" />
-->

  <param name="/use_sim_time" value="false"/>
  <!-- send pr2lite urdf to param server -->
  <param name="robot_description" command="cat '$(find pr2lite_moveit_config)/config/pr2lite_urdf.xml'" />

  <include file="$(find pr2lite_moveit_config)/launch/arbotix_arm.launch" />
  <include file="$(find pr2lite_moveit_config)/launch/velo_gripper.launch" />
  <include file="$(find pr2lite_moveit_config)/launch/point_head.launch" />

<!--
  <include file="$(find pr2lite_moveit_config)/launch/torso_follow_controller.launch" />
  <node name="follow_joint_trajectory" pkg="pr2lite_moveit_config" type="follow_controller.py" ns="arm_controller" output="screen" />
-->
  <node name="statepub" pkg="robot_state_publisher" type="state_publisher" />

  <!-- Monitoring and Diagnostics -->
  <node pkg="maxwell_defs" type="laptop_battery.py" name="laptop_battery" />
  <node pkg="diagnostic_aggregator" type="aggregator_node" name="diagnostic_aggregator" >
      <rosparam command="load" file="$(find pr2lite_moveit_config)/config/pr2lite_diagnostics.yaml" />
  </node>
  <include file="$(find pr2lite_actuators)/launch/linact_server.launch" />
  <node name="rob_st_pub" pkg="robot_state_publisher" type="state_publisher" >
      <param name="publish_frequency" value="5.0" />
  </node>

  <include file="$(find pr2lite_moveit_config)/launch/fake_pub.launch" />

  <include file="$(find pr2lite_nav)/launch/nav_amcl.launch" /> 

  <node pkg="prlite_tf" type="tf_broadcaster" name="transform_configuration" output="screen"/>

  <node pkg="pr2lite_moveit_config" type="joint_states_listener.py" name="joint_states_listener" />

  <include file="$(find pr2lite_moveit_config)/launch/move_group.launch">
    <arg name="allow_trajectory_execution" value="true"/>
  </include>
<!--

  <include file="$(find pr2lite_moveit_config)/launch/moveit_rviz.launch"/>
-->
  <include file="$(find pr2lite_moveit_config)/launch/pr2lite_torso_teleop.launch"/>

</launch>

